sdk/java/pulumi/src/main/java/com/pulumi/experimental/automation/LocalWorkspace.java

```diff
@@ -0,0 +1,1115 @@
+                }
+            }
+        }
+    }
+
+    private void initializeProjectSettings(ProjectSettings settings) throws AutomationException {
+        // If given project settings, we want to write them out to
+        // the working dir. We do not want to override existing
+        // settings with default settings though.
+
+        var existingSettings = getProjectSettings();
+        if (!existingSettings.isPresent()) {
+            saveProjectSettings(settings);
+        } else if (!existingSettings.get().isDefault() && !existingSettings.get().equals(settings)) {
+            var path = findSettingsFile();
+            throw new ProjectSettingsConflictException(path.toString());
+        }
+    }
+
+    /**
+     * Creates a workspace using the specified options. Used for maximal control and
+     * customization of the underlying environment before any stacks are created or
+     * selected.
+     *
+     * @return the workspace
+     * @throws AutomationException if an error occurs
+     */
+    public static LocalWorkspace create() throws AutomationException {
+        return create(null);
+    }
+
+    /**
+     * Creates a workspace using the specified options. Used for maximal control and
+     * customization of the underlying environment before any stacks are created or
+     * selected.
+     *
+     * @param options Options used to configure the workspace
+     * @return the workspace
+     * @throws AutomationException if an error occurs
+     */
+    public static LocalWorkspace create(@Nullable LocalWorkspaceOptions options) throws AutomationException {
+        var cmd = getOrCreatePulumiCommand(options);
+        return new LocalWorkspace(cmd, options);
+    }
+
+    private static PulumiCommand getOrCreatePulumiCommand(
+            @Nullable LocalWorkspaceOptions options) throws AutomationException {
+        return options != null && options.getPulumiCommand() != null
+                ? options.getPulumiCommand()
+                : LocalPulumiCommand.create(LocalPulumiCommandOptions.builder()
+                        .skipVersionCheck(optOutOfVersionCheck(options != null
+                                ? options.getEnvironmentVariables()
+                                : null))
+                        .build());
+    }
+
+    private static boolean optOutOfVersionCheck(@Nullable Map<String, String> environmentVariables) {
+        var hasSkipEnvVar = environmentVariables != null &&
+                environmentVariables.containsKey(LocalPulumiCommand.SKIP_VERSION_CHECK_VAR);
+        var optOut = hasSkipEnvVar || System.getenv(LocalPulumiCommand.SKIP_VERSION_CHECK_VAR) != null;
+        return optOut;
+    }
+
+    /**
+     * Creates a stack with a {@link LocalWorkspace} utilizing the specified inline
+     * (in process) {@code program}. This program is fully debuggable and runs in
+     * process. If no {@link LocalWorkspaceOptions#getProjectSettings()} option is
```

Comment:
Again this comment is wrong I think -- no options here. There may be others too; I've probably missed some overloads. These kinds of things I don't mind leaving to post though.